Bazel CI: //tests/integration/reproducibility:reproducibility_test broken with Bazel@HEAD

https://buildkite.com/bazel/bazel-at-head-plus-downstream/builds/2268#cf713d45-245a-437a-bb0f-85a10498d523
```
exec ${PAGER:-/usr/bin/less} "$0" || exit 1
Executing tests from //tests/integration/reproducibility:reproducibility_test
-----------------------------------------------------------------------------
--- FAIL: Test (35.40s)
    reproducibility_test.go:226: open /private/var/tmp/_bazel_buildkite/151f56892d8b39575ff82475bb0e1262/bazel_testing/bazel_go_test/main0/bazel-out/host/bin/external/go_sdk/builder: no such file or directory
FAIL
```

A bisect points to https://github.com/bazelbuild/bazel/commit/4b47d6f849bb2eca669f48cfdb5d1796100920f3, which looks like  an intentional change.

--- a/tests/integration/reproducibility/reproducibility_test.go
+++ b/tests/integration/reproducibility/reproducibility_test.go
@@ -181,7 +181,11 @@ func Test(t *testing.T) {
 	for _, dir := range dirs {
 		go func(dir string) {
 			defer wg.Done()
-			cmd := bazel_testing.BazelCmd("build", "//:all", "@io_bazel_rules_go//go/tools/builders:go_path")
+			cmd := bazel_testing.BazelCmd("build",
+				"//:all",
+				"@io_bazel_rules_go//go/tools/builders:go_path",
+				"@go_sdk//:builder",
+			)
 			cmd.Dir = dir
 			if err := cmd.Run(); err != nil {
 				t.Fatalf("in %s, error running %s: %v", dir, strings.Join(cmd.Args, " "), err)
@@ -221,7 +225,7 @@ func Test(t *testing.T) {
 	// nominally different per workspace (but in these tests, the go_sdk paths are all set to the same
 	// path in WORKSPACE) -- so if this path is in the builder binary, then builds between workspaces
 	// would be partially non cacheable.
-	builder_file, err := os.Open(filepath.Join(dirs[0], "bazel-out", "host", "bin", "external", "go_sdk", "builder"))
+	builder_file, err := os.Open(filepath.Join(dirs[0], "bazel-bin", "external", "go_sdk", "builder"))
 	if err != nil {
 		t.Fatal(err)
 	}

